### PR TITLE
Modal/Popover update - migrate Modal/Popover to the RN's Modal

### LINKED ISSUE #1681: vibration on Modal !?
## 💬 Question
I have a strange problem:
My modal code is like:

<Modal
        animationType="slide"
        transparent={true}
        visible={modal2Visible2}
        onBackdropPress={() => setModal2Visible(!modal2Visible)}
        backdropStyle={styles.backdrop}
		onRequestClose={() => {
          setModal2Visible(!modal2Visible);
        }}
      >	
<KeyboardAvoidingView      behavior={Platform.OS === "ios" ? "padding" : "height"}    >
<Animated.View style={{opacity: values, transform: [{ scaleY: values }], backgroundColor: "#ffefbf", borderRadius: 20, minWidth: 312}}> 
<ImageBackground    source={require('../../Image/honeycomb2.png')} resizeMode="repeat" style={styles.image}   imageStyle={{ borderRadius: 20}}>
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
		///// other staf
          </View>
        </View>
</ImageBackground>
</Animated.View>
</KeyboardAvoidingView>
</Modal>  

and on some devices (Note 8, android 9), when I have {minWidth: 300} on my Animated.View, (or even as a simple View), I see a strange vibration (as little as a pixel!) in opened modal!
if I set it to my minWidth: 312 or 299 or 301, or anything except 300, It does not vibrate!

I upgraded and downgraded RN and UI Kitten and every thing. Nothing changes.

Why is this happening?

My current setup:
    "react-native": "0.70.5",
    "@eva-design/eva": "^2.1.1",
    "react-native-ui-kitten": "^4.4.1",
    "@ui-kitten/eva-icons": "^5.1.2",

### PR DESCRIPTION
**Pull request by sayres**
### Please read and mark the following check list before creating a pull request:

 - [] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:

Modal and Popover components moved to RN's Modal component implementation

### CODE REVIEW

File: src/components/ui/autocomplete/autocomplete.component.tsx [hunk 1]
```diff
@@ -160,13 +148,21 @@ export class Autocomplete extends React.Component<AutocompleteProps, State> {
     this.props.onFocus && this.props.onFocus(event);
   };
 
+  private onInputFocusAnchor = (event: NativeSyntheticEvent<TextInputFocusEventData>): void => {
```
Comment (malashkevich) [inline src/components/ui/autocomplete/autocomplete.component.tsx:151]: onAncorInputFocus

File: src/components/ui/autocomplete/autocomplete.component.tsx [hunk 2]
```diff
@@ -160,13 +148,21 @@ export class Autocomplete extends React.Component<AutocompleteProps, State> {
...
+    this.inputRefAnchor.current?.blur();
```
Comment (malashkevich) [inline src/components/ui/autocomplete/autocomplete.component.tsx:152]: probably not needed

File: src/components/ui/modal/modal.component.tsx [hunk 3]
```diff
@@ -70,83 +68,38 @@ export class Modal extends React.PureComponent<ModalProps, State> {
...
+        ...state,
+        contentPosition: Point.outscreen(),
+      };
     }
+    return null;
   }
 
   private onContentMeasure = (contentFrame: Frame): void => {
-    this.state.contentFrame = contentFrame;
+    this.setState({contentFrame});
```
Comment (malashkevich) [inline src/components/ui/modal/modal.component.tsx:99]: single call `setState`

File: src/components/theme/modal/modal.service.tsx [hunk 4]
```diff
@@ -94,35 +46,8 @@ import {
  */
 class ModalServiceType {
```
Comment (malashkevich) [inline src/components/theme/modal/modal.service.tsx:47]: Do we need this one as we are using react-native Modal?

File: src/components/ui/modal/modal.component.tsx [hunk 5]
```diff
@@ -11,49 +11,69 @@ import {
...
-} from '../../theme';
+import { ModalService } from '../../theme';
+import {Backdrop, BackdropPresentingConfig} from '@ui-kitten/components/theme/backdrop/backdrop.component';
 
-export interface ModalProps extends ViewProps, ModalPresentingConfig {
+export interface ModalProps extends ViewProps, BackdropPresentingConfig {
   visible?: boolean;
+  shouldUseContainer?: boolean;
   children?: React.ReactNode;
+  animationType?: 'none' | 'slide' | 'fade';
```
Comment (malashkevich) [inline src/components/ui/modal/modal.component.tsx:30]: Please extract to the type

File: src/components/ui/modal/modal.component.tsx [hunk 6]
```diff
@@ -11,49 +11,69 @@ import {
...
+  onShow?: (event: NativeSyntheticEvent<any>) => void;
+  supportedOrientations?: Array<
+    'portrait' | 'portrait-upside-down' | 'landscape' | 'landscape-left' | 'landscape-right'
```
Comment (malashkevich) [inline src/components/ui/modal/modal.component.tsx:33]: Please extract to the type

File: src/components/ui/popover/popover.component.tsx [hunk 7]
```diff
@@ -35,7 +30,7 @@ import {
 
 type PopoverModalProps = Overwrite<ModalProps, {
   children?: React.ReactElement;
-}>;
+}>  & Pick<ModalProps, 'animationType' | 'hardwareAccelerated' | 'supportedOrientations' | 'onShow'>;
```
Comment (malashkevich) [inline src/components/ui/popover/popover.component.tsx:33]: Please extract to the type

File: src/components/ui/popover/type.ts [hunk 8]
```diff
@@ -594,6 +594,124 @@ export class PopoverPlacements {
       return fitsBottom(frame, other) && fitsLeft(frame, other) && fitsRight(frame, other);
     }
   };
+  static INNER: PopoverPlacement = new class implements PopoverPlacement {
```
Comment (malashkevich) [inline src/components/ui/popover/type.ts:597]: It does makes sense to split this file into multiple

**Review by malashkevich — APPROVED** [review]
A few comments, but looks good

File: src/components/theme/application/applicationProvider.component.tsx [hunk 9]
```diff
@@ -14,26 +14,26 @@ import {
 } from '@eva-design/dss';
 import { StyleProvider } from '../style/styleProvider.component';
 import { ThemeProviderProps } from '../theme/themeProvider.component';
-import { ModalPanel } from '../modal/modalPanel.component';
- 
+// import { ModalPanel } from '../modal/modalPanel.component';
```
Comment (flexbox) [inline src/components/theme/application/applicationProvider.component.tsx:17]: I don't this you want to leave this comment here :)

File: src/components/ui/autocomplete/autocomplete.component.tsx [hunk 10]
```diff
@@ -190,12 +190,30 @@ export class Autocomplete extends React.Component<AutocompleteProps, State> {
...
+    );
+  };
+
   private renderInputElement = (props: InputProps): InputElement => {
     return (
       <View>
         <Input
           {...props}
           ref={this.inputRef}
+          testID='@autocomplete/input'
```
Comment (greenfrvr) [inline src/components/ui/autocomplete/autocomplete.component.tsx:214]: Shouldn't we allow users to define this `testID` themselves? I'm concerned they won't be able to run tests correctly with such `testID` definition. Inputs are usually essential component for test cases.

**Comment by sayres** [comment]
fixes #1585

File: src/components/ui/popover/popover.component.tsx [hunk 11]
```diff
@@ -241,13 +205,27 @@ export class Popover extends React.Component<PopoverProps, State> {
...
   public render(): React.ReactElement {
     return (
-      <MeasureElement
-        shouldUseTopInsets={ModalService.getShouldUseTopInsets}
-        force={this.state.forceMeasure}
-        onMeasure={this.onChildMeasure}
-      >
-        {this.props.anchor()}
-      </MeasureElement>
+      <View>
```
Comment (siarheipashkevich) [inline src/components/ui/popover/popover.component.tsx:208]: Why did you add this `View` component. It adds additional wrapper which broke some ui.

Comment (siarheipashkevich) [reply src/components/ui/popover/popover.component.tsx:208]: @sayres @malashkevich please check my comment

Comment (siarheipashkevich) [reply src/components/ui/popover/popover.component.tsx:208]: maybe there is should be `Fragment` ?
